raft/src/test/java/org/apache/kafka/raft/SimpleKeyValueStoreTest.java

```diff
@@ -59,18 +63,17 @@ private KafkaRaftClient setupSingleNodeRaftManager() {
 
     @Test
     public void testPutAndGet() throws Exception {
-        KafkaRaftClient manager = setupSingleNodeRaftManager();
-        manager.initialize(new NoOpStateMachine());
-        SimpleKeyValueStore<Integer, Integer> store = new SimpleKeyValueStore<>(manager,
+        SimpleKeyValueStore<Integer, Integer> store = new SimpleKeyValueStore<>(
             new Serdes.IntegerSerde(), new Serdes.IntegerSerde());
-        store.initialize();
+        KafkaRaftClient manager = setupSingleNodeRaftManager();
+        manager.initialize(store);
 
         CompletableFuture<OffsetAndEpoch> future = store.put(0, 1);
         manager.poll();
 
         assertTrue(future.isDone());
         // The control record takes up one offset.
-        assertEquals(new OffsetAndEpoch(2L, 1), future.get());
+        assertEquals(new OffsetAndEpoch(1L, epoch), future.get());
```

Comment:
We fixed one implicit issue of leader change message appended twice when we call both `manager.initialize(new NoOpStateMachine())` and `store.initialize()` at the same time.